File: src/main/java/ru/citeck/idea/index/EcosFileIndexer.java
package ru.citeck.idea.index;

import com.intellij.openapi.extensions.ExtensionPointName;
import com.intellij.util.indexing.FileContent;
import org.jetbrains.annotations.NotNull;

import ru.citeck.idea.files.FileType;
import ru.citeck.idea.files.FileTypeService;

public interface EcosFileIndexer {

    ExtensionPointName<EcosFileIndexer> EP_NAME = ExtensionPointName.create("ru.citeck.idea.fileIndexer");

    boolean accept(FileType fileType);

    void map(@NotNull FileContent inputData, Indexes.FileIndexes indexes);

    default FileType getFileType(FileContent inputData) {
        return FileTypeService.getInstance()
            .getFileType(inputData.getFile(), inputData.getProject());
    }

}

File: src/main/kotlin/ru/citeck/idea/search/index/EcosFileIndexer.kt
package ru.citeck.idea.search.index

import com.intellij.openapi.extensions.ExtensionPointName
import com.intellij.util.indexing.FileContent
import ru.citeck.idea.files.FileType
import ru.citeck.idea.files.FileTypeService
import ru.citeck.idea.search.index.Indexes.FileIndexes

interface EcosFileIndexer {

    companion object {
        val EP_NAME: ExtensionPointName<EcosFileIndexer> = ExtensionPointName.create("ru.citeck.idea.fileIndexer")
    }

    fun accept(fileType: FileType): Boolean

    fun map(inputData: FileContent, indexes: FileIndexes)

    fun getFileType(inputData: FileContent): FileType {
        return FileTypeService.getInstance()
            .getFileType(inputData.file, inputData.project)
    }
}
